ooni/templates/httpt.py

```diff
@@ -122,11 +122,28 @@ def addToReport(self, request, response=None, response_body=None, failure_string
 
             failure (instance): An instance of :class:twisted.internet.failure.Failure
         """
+        def _representHeaders(headers):
+            represented_headers = {}
+            for name, value in headers.getAllRawHeaders():
+                represented_headers[name] = value[0]
+            return represented_headers
+
+        def _representBody(body):
+            # XXX perhaps add support for decoding gzip in the future.
+            try:
+                body.replace('\0', '')
```

Comment:
Oh my you are right that is a very good point.

The rational behind this is that if we end up with bodies with null bytes inside of them this will lead to some JSON parsers from incorrectly parsing the data. Since it's not a big loss to loose null characters this is being added here.

I guess it should probably happen after we have determined that it's not 'ascii' or 'unicode' though since a null byte may actually mean something in a non ascii or utf-8 encoding.

I will push a fix to this.